ironfish/src/migrations/data/013-wallet-2.ts

```diff
@@ -44,427 +59,482 @@ export class Migration013 extends Migration {
-        await metaStoreNew.put('defaultAccountId', null, tx)
+      if (!transaction) {
+        countMissingTx++
+        speed.add(1)
+        continue
       }
-    }
 
-    await metaStoreOld.del('defaultAccountName', tx)
-    await metaStoreOld.del('headHash', tx)
-  }
+      const encryptedNote = findNoteInTransaction(transaction.transaction, noteHash)
+      Assert.isNotNull(encryptedNote)
 
-  async migrateAccounts(
-    accountsStoreOld: Stores['old']['accounts'],
-    accountsStoreNew: Stores['new']['accounts'],
-    tx: IDatabaseTransaction,
-    logger: Logger,
-  ): Promise<{ account: DatabaseStoreValue<NewStores['accounts']>; id: string }[]> {
-    const accounts = []
+      let ownerAccount: DatabaseStoreValue<Stores['new']['accounts']> | null = null
+      let ownerNote: Note | null = null
+
+      for (const possibleAccount of accounts) {
+        const received = encryptedNote.decryptNoteForOwner(possibleAccount.incomingViewKey)
+
+        if (received) {
+          ownerAccount = possibleAccount
+          ownerNote = received
+          break
+        }
+      }
+
+      if (!ownerAccount || !ownerNote) {
+        logger.warn(
+          `\tCould not find the original account that the note ${noteHashHex} was decrypted as owner, discarding. Tried ${accounts.length} accounts.`,
+        )
+        countMissingAccount++
+        speed.add(1)
+        continue
+      }
+
+      const decryptedNote = {
+        accountId: ownerAccount.id,
+        noteIndex: nullifierEntry.noteIndex,
+        serializedNote: ownerNote.serialize(),
+        spent: nullifierEntry.spent,
+        transactionHash: transactionHash,
+        nullifierHash: nullifierEntry.nullifierHash
+          ? Buffer.from(nullifierEntry.nullifierHash, 'hex')
+          : null,
+      }
+
+      const accountPrefix = accountPrefixes.get(ownerAccount.id)
+      Assert.isNotUndefined(accountPrefix)
+
+      // These writes must happen atomically
+      await accountsDb.withTransaction(tx, async (tx) => {
+        Assert.isNotNull(ownerAccount)
+        Assert.isNotNull(ownerNote)
+
+        await stores.new.decryptedNotes.put([accountPrefix, noteHash], decryptedNote, tx)
+        await stores.new.transactions.put([accountPrefix, transactionHash], transaction, tx)
+
+        if (!decryptedNote.spent) {
+          let balance = unconfirmedBalances.get(ownerAccount.id) ?? BigInt(0)
+          balance += ownerNote.value()
+          unconfirmedBalances.set(ownerAccount.id, balance)
+          await stores.new.balances.put(ownerAccount.id, balance, tx)
+        }
+
+        if (decryptedNote.nullifierHash) {
+          await stores.new.nullifierToNoteHash.put(
+            [accountPrefix, decryptedNote.nullifierHash],
+            noteHash,
+            tx,
+          )
+        }
 
-    for await (const [accountName, accountValue] of accountsStoreOld.getAllIter(tx)) {
-      const accountId = uuid()
+        await stores.old.noteToNullifier.del(noteHashHex, tx)
```

Comment:
```suggestion
        // Delete note from old store so that it will not be reprocessed if we resume the migration
        await stores.old.noteToNullifier.del(noteHashHex, tx)
```

Reply:
This is subtle but important to make resuming migrations work, so I wanted to call it out